lib/src/facts/resolvers/operating_system_resolver.cc

```diff
@@ -57,6 +57,18 @@ namespace facter { namespace facts { namespace resolvers {
         if (!data.release.empty()) {
             auto value = make_value<map_value>();
 
+            // When we have no major or minor, do a 'trivial' parse of
+            // the release to try to get SOMETHING out of it.
+            if(data.minor.empty() && data.major.empty()) {
+                 auto begin = data.release.begin();
+                 auto end = data.release.end();
+                 auto end_first = find(begin, end, '.');
+                 auto begin_second = next(end_first);
+                 auto end_second = find(begin_second, end, '.');
+                 data.major = std::string(begin, end_first);
+                 data.minor = std::string(begin_second, end_second);
```

Comment:
Ah, that uses indexes instead of iterators. I'm not particular dead-set on iterators here, so could definitely use that code instead of we think it's more readable (and honestly it probably is).